src/components/chat/chat.js

```diff
@@ -0,0 +1,241 @@
+      sessionStorage.setItem('timeStarted', Date.now());
+    }
+
+    if (sessionStorage.getItem('timeStarted')) {
+      const newTime = Date.now() - sessionStorage.getItem('timeStarted');
+      const secs = Number(((newTime % 60000) / 1000).toFixed(0));
+      const mins= Math.floor(newTime/ 60000);
+      this.setState({
+        minutes: mins,
+        seconds: secs,
+      })
+    }
+
+    const intervalId = setInterval( () => {
+      this.setState({seconds: this.state.seconds + 1})
+      if (this.state.seconds < 10 ) this.setState({secondsString: '0' + this.state.seconds})
+      else this.setState({secondsString: this.state.seconds})
+      if (this.state.seconds === 60) {
+        this.setState({ seconds: 0, minutes: this.state.minutes + 1, secondsString: '00'})
+      }
+      if (this.state.minutes === 15) {
+        this.setState({ overTime: 'red'});
+      }
+    }, 1000)
+
+    this.setState({ timerId: intervalId });
+
+  }
+
+  renderOverlay = () => {
+    if (this.state.tutorOrLearner === 'learner' && !this.state.tutorJoined) {
+      return <Overlay closeOverlay={(counter) => {
+        clearInterval(counter);
+        if (this.props.question.learner) { 
+          // prevents chat from crashing when the timer runs out
+          const targetOffer = this.props.offers.filter(offer => offer.offer_id === this.props.question.answered_by);
+          this.props.history.goBack();
+          this.props.socket.emit('cancel call', targetOffer[0].tutor);
+        }
+      }
+      }/>
+    }
+  }
+
+  hangUp = () => {
+    this.props.socket.emit('hang up', {roomId: this.state.roomId});
+  }
+
+  updateKarma = (karma) => {
+    this.props.socket.emit('update karma', {tutor: this.state.targetTutor, karma: karma });
+  }
+
+  showEndChatModal = () => {
+    if (this.state.showFeedbackModal) {
+      sessionStorage.removeItem('timeStarted');
+      sessionStorage.removeItem('targetOffer');
+      sessionStorage.removeItem('chatDetails');
+      return <ModalEndChat showModal={this.state.showFeedbackModal} closeQuestion={(question) => this.props.updateQuestionStatus(question)} updateKarma={this.updateKarma} closeChatModal={() => this.setState({showFeedbackModal: false})} history={this.props.history} questionId={this.state.questionId} tutorOrLearner={this.state.tutorOrLearner}/>
+    }
+  }
+
+  showDetails = () => {
+    if (this.state.showMoreInfo) {
+      return <ShowDetails all={this.state} closeDetailsModal={() => this.setState({showMoreInfo: false})}/>
+    }
+  }
+
+  componentWillUnmount() {
+    this.props.socket.removeListener('join room');
+    this.props.socket.removeListener('question info');
+    this.props.socket.removeListener('hang up');
+    clearInterval(this.state.timerId);
+  }
+
+  render() {
+    return(
+      <div className="chat-component">
+
+        {this.state.tutorJoined ? null : this.renderOverlay()}
+
+        <div className="chat-header">
```

Comment:
As this component is quite large, I recommend you not rendering any specific parts, but to delegate in other presentational components. You did it correctly with the rest of sections in this render function but not with this header.